src/rng/src/main.rs

```diff
@@ -0,0 +1,192 @@
+// VIRTIO RNG Emulation via vhost-user
+//
+// Copyright 2021 Linaro Ltd. All Rights Reserved.
+// Mathieu Poirier <mathieu.poirier@linaro.org>
+//
+// SPDX-License-Identifier: Apache-2.0
+
+mod vhu_rng;
+
+use clap::{load_yaml, App};
+use std::fs::File;
+use std::path::Path;
+use std::sync::{Arc, Mutex, RwLock};
+use std::thread;
+use vhost::{vhost_user, vhost_user::Listener};
+use vhost_user_backend::VhostUserDaemon;
+use vhu_rng::VuRngBackend;
+use vm_memory::{GuestMemoryAtomic, GuestMemoryMmap};
+
+fn start_backend(cmd_args: clap::ArgMatches, dry_run: bool) -> Result<(), String> {
```

Comment:
This function can be split in 2 high level components to make testing easier:
1. Parsing the command line arguments, and making sure they're valid => this function should return a new object type representing the device configuration
2. Based on the configuration, start a backend. Probably the backend can further done be split in multiple components, but we can work on this incrementally.

Reply:
Once we do the split, it will become more obvious how to get rid of the `dry_run` parameter.

Reply:
Things have been split along the lines you requested and the dry_run parameter naturally goes away.  I also moved the CLI parsing and backend functionality to different files to help with testing.